src/gax/src/retry_policy.rs

```diff
@@ -587,16 +624,15 @@ where
         }
     }
 
-    fn on_throttle(&self, loop_start: std::time::Instant, attempt_count: u32) -> Option<Error> {
-        if let Some(e) = self.inner.on_throttle(loop_start, attempt_count) {
-            return Some(e);
-        }
-        if attempt_count < self.maximum_attempts {
-            None
-        } else {
-            Some(Error::exhausted(LimitedAttemptCountError::new(
-                self.maximum_attempts,
-            )))
+    fn on_throttle(
+        &self,
+        loop_start: std::time::Instant,
+        attempt_count: u32,
+        error: Error,
+    ) -> ThrottleResult {
+        match self.inner.on_throttle(loop_start, attempt_count, error) {
+            ThrottleResult::Continue(e) => self.stop_if_exhausted(attempt_count, e),
+            ThrottleResult::Exhausted(e) => ThrottleResult::Exhausted(e),
         }
```

Comment:
optional nit: I think we can just return `self.inner.on_throttle(...)`

If we have hit the maximum attempt count, the retry loop was terminated before the throttler gets involved.

(Attempts do not increase on throttle).